### DISCUSSION TITLE
Is it possible to store the zone in database (Mysql)

### DISCUSSION THREAD

**Discussion by the-yaz** [annotorious/annotorious#313] [Q&A]
I implemented the annotorious and it's working fine until now , the user can upload an image and then add zones to it , but want to store the zones so whenever he open again the image he should find the zones that he defines, is it possible to do it in mysql ? unfortunately I didn't find enough explanation in the documentation  about storing the zones or annototions in general .

**Comment by rsimon** [comment]
You can use any backend you want. But it will be up to you to build the necessary web application. Annotorious handles only the frontend part. In a nutshell, you'll need to use the [events](https://annotorious.github.io/api-docs/annotorious/#events) (most importantly `createAnnotation`, `deleteAnnotation` and `updateAnnotation`) and then write your own JavaScript that POSTs the data to your backend.

You'll also need an endpoint that retrieves all the annotations for the current image, and then load these when your application starts. (And then use `anno.setAnnotations` to display them.)

**Reply by the-yaz** [reply]
is there any example of storing annotation in any kind of database ?

**Reply by rsimon** [reply]
Something like this might help you get started:
https://blog.logrocket.com/build-rest-api-node-express-mysql/
